lib/config/set-envs.js

```diff
@@ -0,0 +1,58 @@
+// Set environment variables for any non-default configs,
+// so that they're already there when we run lifecycle scripts.
+//
+// See https://github.com/npm/rfcs/pull/90
+
+const envName = name => `npm_config_${name.replace(/-/g, '_')}`
```

Comment:
nit: would be nice if we had a rule of thumb that any time you use regexp, no matter how simple, you have to comment what you're intent is with it.

Reply:
In more elaborate cases, I might agree, but as a rule, that seems to just tend towards excessive commenting.

```js
// replaces - with _
str.replace(/-/g, '_')
```

which is in the same category as:

```js
// adds 1 to foo
foo += 1
```

Why would we need this for trivial regexps and not for every other trivial thing?

Reply:
I agree with this.